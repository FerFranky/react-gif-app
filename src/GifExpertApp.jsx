import { useState } from "react"
import { AddCategory, GifGrid } from "./components"

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon ball'])
    const onAddCategory = (value) => {
        if (categories.includes(value)) return
        setCategories([value, ...categories])
    }
    return (
        <>
            <h1>GifExpertApp</h1>
            
            <AddCategory onAddCategory={ (value) => onAddCategory(value) } />
            {categories.map(category => ( <GifGrid key={category} category={category} /> ))}
        </>
    )
}