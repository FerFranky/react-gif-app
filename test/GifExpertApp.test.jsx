import { GifExpertApp } from "../src/GifExpertApp"
import { render, screen } from "@testing-library/react"

describe('Pruebas de componente <GifExpertApp />', () => { 
    const titleApp = 'GifExpertApp'

    test('Debe coincidir con el snapshot', () => { 
        const { container } = render(<GifExpertApp />)
        expect(container).toMatchSnapshot()
     })

     test('Debe mostrar el titulo de la aplicacion', () => { 
        render(<GifExpertApp />)
        expect( screen.getByText(titleApp)).toBeTruthy()
     })
 })