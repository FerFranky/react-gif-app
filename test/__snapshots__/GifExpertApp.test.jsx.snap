// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente <GifExpertApp /> Debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One Punch
  </h3>
  <h3>
    Cargando...
  </h3>
  <h3>
    Dragon ball
  </h3>
  <h3>
    Cargando...
  </h3>
</div>
`;
